mechafil/data_spacescope.py

```diff
@@ -0,0 +1,280 @@
+SPACESCOPE_AUTH_KEY = 'Bearer ghp_xJtTSVcNRJINLWMmfDangcIFCjqPUNZenoVe'
+
+def get_historical_network_stats(
+    start_date: datetime.date, current_date: datetime.date, end_date: datetime.date
+) -> pd.DataFrame:
+    power_df = query_spacescope_power_stats(start_date, current_date)
+    onboards_df = query_spacescope_daily_power_onboarded(start_date, current_date)
+    stats_df = query_spacescope_supply_stats(start_date, current_date)
+    stats_df = stats_df.merge(power_df, on="date", how="inner").merge(
+        onboards_df, on="date", how="inner"
+    )
+    renewal_df = get_day_renewed_power_stats(
+        start_date, current_date, end_date
+    )
+    stats_df = stats_df.merge(renewal_df, on='date', how='inner')
+    return stats_df
+
+
+def get_sector_expiration_stats(
+    start_date: datetime.date,
+    current_date: datetime.date,
+    end_date: datetime.date,
+) -> pd.DataFrame:
+    scheduled_df = query_spacescope_sector_expirations(start_date, end_date)
+    filter_scheduled_df = scheduled_df[
+        scheduled_df["date"] >= pd.to_datetime(current_date, utc="UTC")
+    ]
+    rbp_expire_vec = filter_scheduled_df["total_rb"].values
+    qap_expire_vec = filter_scheduled_df["total_qa"].values
+    # we need the entire history of known_scheduled_pledge_release, so get the
+    # data from the entire time-window, not just from current-date onwards
+    pledge_release_vec = scheduled_df["total_pledge"].values
+    return rbp_expire_vec, qap_expire_vec, pledge_release_vec
+
+
+def get_day_renewed_power_stats(
+    start_date: datetime.date,
+    current_date: datetime.date,
+    end_date: datetime.date,
+) -> Tuple[np.array, np.array]:
+    scheduled_df = query_spacescope_sector_expirations(start_date, end_date)
+    filter_scheduled_df = scheduled_df[
+        scheduled_df["date"] <= pd.to_datetime(current_date, utc="UTC")
+    ]
+    rb_renewal_rate = (
+        filter_scheduled_df["extended_rb"] / filter_scheduled_df["total_rb"]
+    ).values
+    day_renewed_qa_power = filter_scheduled_df["extended_qa"].values
+    renewal_df = pd.DataFrame(
+        {
+            'date': pd.to_datetime(filter_scheduled_df['date']).dt.date,
+            'rb_renewal_rate': rb_renewal_rate,
+            'day_renewed_qa_power_pib': day_renewed_qa_power
+        }
+    )
+    return renewal_df
+
+
+def chunk_dates(start_date: datetime.date, 
+                end_date: datetime.date,
+                chunks_days: int = DEFAULT_SPACESCOPE_CHUNK_SIZE_IN_DAYS) -> List:
+    chunk_start = start_date
+    dates_chunked = []
+    while chunk_start <= end_date:
+        chunk_end = min(chunk_start + datetime.timedelta(days=chunks_days), end_date)
+        dates_chunked.append((chunk_start, chunk_end))
+
+        chunk_start = chunk_end + datetime.timedelta(days=1)
+
+    return dates_chunked
+
+def query_spacescope_sector_expirations(
+    start_date: datetime.date, 
+    end_date: datetime.date,
+    chunk_days: int = DEFAULT_SPACESCOPE_CHUNK_SIZE_IN_DAYS
```

Comment:
`chunk_days` is not used in this function. It can be removed for simplicity.

Reply:
done